Add dev package to docs (#2578)
DataDog/integrations-core: docs/index.rst

The Datadog Check Toolkit
-------------------------

The :ref:`datadog_checks_base` package provides a set of functionalities used by any Agent based Integration:

* :class:`~datadog_checks.base.checks.base.AgentCheck`, the base class that every Agent-based Integration is derived from.
* :ref:`Prometheus <prometheus>` and :ref:`OpenMetrics <openmetrics>` facilities.
* Boilerplate code implementing common operations.
* A testing framework that can be used independently of the Agent.

> The :ref:`datadog_checks_dev` package provides tools to ease development and testing:

Reviewer: ```suggestion
The :ref:`datadog_checks_dev` package provides tools for use during development and testing:
```

"Ease" suggests that dev and testing are a negative condition that needs to be treated with painkillers or something. :)
